app/controllers/welcome_controller.rb

```diff
@@ -1,4 +1,11 @@
+# frozen_string_literal: true
+
+require 'csv'
+
 class WelcomeController < ApplicationController
   def index
+    @shows ||= CSV.parse(File.read('netflix_titles.csv'), headers: true, header_converters: :symbol)
```

Comment:
this will still load the whole file into memory and then you will iterate only first x amount. What you could do is something like this:
`CSV.foreach('netflix_titles.csv', headers: true, header_converters: :symbol).take((params[:limit] || 10).to_i)`

This would only grab first n lines from the csv file and then in html file you can just iterate over shows like:

@shows.each do |show|
   ...insert html code here...
end